hs_core/tests/api/views/test_authorize.py

```diff
@@ -0,0 +1,853 @@
+
+        self._run_tests(self.request, parameters)
+
+        # test for public resource
+        self.assertTrue(self.res.raccess.discoverable)
+        self.assertFalse(self.res.raccess.public)
+        self.res.raccess.discoverable = False
+        self.res.raccess.public = True
+        self.res.raccess.save()
+
+        self._run_tests(self.request, parameters)
+
+        # test for immutable/published resource
+        self.assertFalse(self.res.raccess.immutable)
+        self.assertFalse(self.res.raccess.published)
+        self.assertTrue(self.res.raccess.public)
+
+        self.res.raccess.immutable = True
+        self.res.raccess.published = True
+        self.res.raccess.save()
+
+        self._run_tests(self.request, parameters)
+
+    def test_authorize_superuser(self):
+        # create super user
+        super_user = users.create_account(
+            'super_user@email.com',
+            username='superuser',
+            first_name='super_first_name',
+            last_name='super_last_name',
+            superuser=True,
+            groups=[])
+
+        self.request.user = super_user
+
+        common_parameters = [
+                      # super user has authorization for resource metadata view for a resource
+                      # that is private or discoverable or public or published
+                      {'res_id': self.res.short_id, 'needed_permission': Action_To_Authorize.VIEW_METADATA,
+                       'success': True, 'exception': None},
+                      # super user has authorization for resource view (both metadata and content files)for a resource
+                      # that is private or discoverable or public or published
+                      {'res_id': self.res.short_id, 'needed_permission': Action_To_Authorize.VIEW_RESOURCE,
+                       'success': True, 'exception': None},
+                      # super user has authorization for editing a resource that is private or discoverable or public
+                      # or published
+                      {'res_id': self.res.short_id, 'needed_permission': Action_To_Authorize.EDIT_RESOURCE,
+                       'success': True, 'exception': None},
+                      # super user has no authorization for creating a resource version
+                      {'res_id': self.res.short_id, 'needed_permission': Action_To_Authorize.CREATE_RESOURCE_VERSION,
+                       'success': False, 'exception': PermissionDenied}
+                      ]
+
+        parameters = [
+                      # super user has authorization for setting resource flags (e.g., public,
+                      # published/immutable, shareable etc) for a private resource
+                      {'res_id': self.res.short_id, 'needed_permission': Action_To_Authorize.SET_RESOURCE_FLAG,
+                       'success': True, 'exception': None},
+                      # super user has authorization for deleting a private resource
```

Comment:
An administrator should have the permissions to delete a published resource, authorization to be only used in extreme situations, but if the site is abused an administrator needs to be able to take a resource down.  Note that I prefer the term "Administrator" to super user in these messages.